src/main/java/org/springframework/integration/aws/config/xml/SqsOutboundChannelAdapterParser.java

```diff
@@ -34,13 +34,33 @@
  */
 public class SqsOutboundChannelAdapterParser extends AbstractOutboundChannelAdapterParser {
 
+    public static final String QUEUE_MESSAGING_TEMPLATE_REF = "queue-messaging-template";
+
+    private boolean hasAttribute(Element element, String attribute) {
+        return StringUtils.hasText(element.getAttribute(attribute));
+    }
+
 	@Override
 	protected AbstractBeanDefinition parseConsumer(Element element, ParserContext parserContext) {
 		BeanDefinitionBuilder builder = BeanDefinitionBuilder.genericBeanDefinition(SqsMessageHandler.class);
-		builder.addConstructorArgReference(element.getAttribute(AmazonWSParserUtils.SQS_REF));
-		if (StringUtils.hasText(element.getAttribute(AmazonWSParserUtils.RESOURCE_ID_RESOLVER_REF))) {
+
+        if (hasAttribute(element, QUEUE_MESSAGING_TEMPLATE_REF) &&
+                (
+                        hasAttribute(element, AmazonWSParserUtils.SQS_REF) ||
+                        hasAttribute(element, AmazonWSParserUtils.RESOURCE_ID_RESOLVER_REF))) {
+            throw new IllegalStateException(QUEUE_MESSAGING_TEMPLATE_REF + " should not be defined in conjunction with " + AmazonWSParserUtils.SQS_REF + " or " + AmazonWSParserUtils.RESOURCE_ID_RESOLVER_REF);
```

Comment:
Hi, @rahulpilani !

The fix looks good.
However let me provide some comments and suggestions anyway.

Here it would be better to use this:
```
parserContext.getReaderContext().error()
```
In addition, since you make `sqs` attr as non-required there is need to add one more check that one of `sqs` or `queue-messaging-template` is present instead of that unclear exception where there is no appropriate empty ctor.

Similar check must be applied for the `SqsUoutboundChannelAdapterParser`, because `sqs` isn't required already.

Appropriate test would be good, too.